### DISCUSSION TITLE
Cast Lists

### DISCUSSION THREAD

**Discussion by ingoboerner** [dracor-org/dracor-api#256] [Ideas]
Maybe we should invest into making cast lists (element `<castList>`) more "useful" (for research), e.g. by linking the individual `<castItem>` to `<person>` in `<particDesc>`.

An example <castList>: 

```
<castList>
        <head>Personen.</head>
        <castItem><role>Emilia Galotti.</role></castItem>
        <castGroup rend="braced">
          <castItem><role>Odoardo,</role></castItem>
          <castItem><role>Claudia Galotti,</role></castItem>
          <roleDesc>Eltern der Emilia.</roleDesc>
        </castGroup>
        <castItem><role>Hettore Gonzaga,</role>
          <roleDesc>Prinz von Guastalla.</roleDesc></castItem>
        <castItem><role>Marinelli,</role>
          <roleDesc>Kammerherr des Prinzen.</roleDesc></castItem>
        <castItem><role>Camillo Rota,</role>
          <roleDesc>einer von des Prinzen Räten.</roleDesc></castItem>
        <castItem><role>Conti,</role>
          <roleDesc>Maler.</roleDesc></castItem>
        <castItem><role>Graf Appiani.</role></castItem>
        <castItem><role>Gräfin Orsina.</role></castItem>
        <castItem><role>Angelo,</role> und <role>einige Bediente.</role></castItem>
      </castList>
```

and the corresponding <particDesc> which is used when creating/labelling/... the network:

```
<particDesc>
        <listPerson>
          <person xml:id="der_prinz" sex="MALE">
            <persName>Der Prinz</persName>
          </person>
          <person xml:id="der_kammerdiener" sex="MALE">
            <persName>Der Kammerdiener</persName>
          </person>
          <person xml:id="conti" sex="MALE">
            <persName>Conti</persName>
          </person>
          <person xml:id="marinelli" sex="MALE">
            <persName>Marinelli</persName>
          </person>
          <person xml:id="camillo_rota" sex="MALE">
            <persName>Camillo Rota</persName>
          </person>
          <person xml:id="claudia" sex="FEMALE">
            <persName>Claudia</persName>
          </person>
          <person xml:id="pirro" sex="MALE">
            <persName>Pirro</persName>
          </person>
          <person xml:id="odoardo" sex="MALE">
            <persName>Odoardo</persName>
          </person>
          <person xml:id="angelo" sex="MALE">
            <persName>Angelo</persName>
          </person>
          <person xml:id="emilia" sex="FEMALE">
            <persName>Emilia</persName>
          </person>
          <person xml:id="appiani" sex="MALE">
            <persName>Appiani</persName>
          </person>
          <person xml:id="battista" sex="MALE">
            <persName>Battista</persName>
          </person>
          <person xml:id="orsina" sex="FEMALE">
            <persName>Orsina</persName>
          </person>
          <listRelation type="personal">
            <relation name="parent_of" active="#odoardo #claudia" passive="#emilia"/>
            <relation name="associated_with" active="#marinelli" passive="#der_prinz"/>
            <relation name="associated_with" active="#camillo_rota" passive="#der_prinz"/>
          </listRelation>
        </listPerson>
      </particDesc>
```

Just some observation/guesses:
* the order of the "items" is different. In the `<particDesc>` "Prinz" is listed as first character; he speaks first; --> `<person>`elements are ordered by order they speak on stage (is this true for all plays, we could check that automatically). In the `castList` though we have a different order (by rank, status, random, how the author wanted it to be presented). To look at this ordering alone would be interesting, I guess.
* The name of a character in a `<castItem>` might be different to the "Ansetzungsform" in the ´<particDesc>´. The "names"/"labels" of the characters in the `<person>`elements might be based on the speaker labels `<speaker>` in the `<text>`, but also might be not... The <castItem> would contain the "name" of a character as intended by the author. At the moment we can not connect that to the individual `<speaker>` labels. Looking at the "name" of a character in the `<castList>` and comparing it to each value in `distinct-values(//speaker[@who eq '{character-id}'])` could be interesting.
* Additional information that might be included in the <castList>, even though it might be marked up, like "traits", "age", ... whatever is not extractable at the moment (I mean, via the API), because the `.../characters` endpoint to my understanding is only taking information from `<person xml:id="{character-id}"`.
* Not all speaking characters <person> might be mentioned in the `<castList>` as `<castItem>`; not all characters  <castItem> in the <castList> might "speak" in the play and thus might not be included in the <particDesc>

What we could do for a start is linking `<castItem>` to the respective `<person>` element with `@corresp` or vice versa, e.g.

1) Use `@corresp` on the `<castItem>`
```
<person xml:id="emilia" sex="FEMALE">
            <persName>Emilia</persName>
</person>
```

```
<castItem @corresp="#emilia"><role>Emilia Galotti.</role></castItem>
```

2) introduce actual ids for the `<castItem>`:

```
<castItem xm:id="{prefix}_emilia"><role>Emilia Galotti.</role></castItem>
```

and then
```
<person xml:id="emilia" sex="FEMALE" corresp="#{prefix}_emilia">
            <persName>Emilia</persName>
</person>
```

Option 2 would involve two edit operations but all relevant information would be connected to the `<person>` element, whereas option 1 is the less intrusive variant (1 edit and we are ready to go).

3) There could be a third option using some kind of `<relation>` but this does not really seem necessary at the moment (could be when wanting to record the type of relation, e.g. unclear, ...?) But maybe not start with things like that.

The question remains how to add this information, e.g. `@corresp` as in option 1.

At the ExploreCor TS I talked to @cmil about maybe re-using his "who is" GUI to make the connections of `<castItem>` and `<person>`? So, we seem to already have some tool that could be re-used for this annotation task..

So, what do you think about it?

(It is also somehow connected to #255)

**Comment by cmil** [comment]
Two quick comments:

1. Technically we probably do not need to link `castItem`s and `person`s in two directions, i.e. with both `@xml:id` and `@corresp` on both sides. What would be the use case for that 2. step of linking that  requires `@xml:id`s on `castItem`s?
2. The epdacor-whois is highly specialised on the epdracor-sources and would have to be heavily modified to work in a more general context. But it can certainly serve as a model for newly developed UI that could handle the easier use case of linking an already existing `particDesc` with `castItem`s.

**Reply by ingoboerner** [reply]
ad 1. I was just listing some options of how to link them, and did not mean, that we need to use all of them together. I would go for `@corresp` on the `<castItem>` and use the `<person>`s `@xml:id` for a start, so the first option.

**Reply by lehkost** [reply]
If we decide to link them, this would be my preferred option, too.

**Comment by ingoboerner** [comment]
```
<listPerson>
<person xml:id="emilia">....</person>
</listPerson>
```

```
<castList>
<castItem corresp="#emilia">gsdfgfd</castItem>
</castList>
```
